### DISCUSSION TITLE
Metrics request complete

### DISCUSSION THREAD

**Discussion by vrshgv** [Unleash/unleash#7051] [Q&A] [answered]
Hi! 
I use `unleash-proxy-client` and after getting a variant with `getVariant` method in browser Network tab there is an POST metrics request `<url>/api/frontend/client/metrics`. And this request does not always fire in time, for example if there is an redirect right after getting a variant. How do I make sure to wait for metrics request to complete after `getVariant` call?

**Comment by ivarconr** [comment, marked as answer]
Hi @vrshgv, 

this happens because the initial metrics call is sent after 2 seconds. In the latest beta release of the JavaScript SDK (`v3.4.0-beta.1`) we added an option that allows you to control when the initial metrics are pushed, via the option called `metricsIntervalInitial`.  By setting it to 1 second it might help some, but probably not be optimal for your use-case. 

I think a better solution for this would be for the SDK to expose a `async sendMetrics()` function so that you could ensure in your application that metrics are sent, before the redirect. Feel free to open a issue request in the repository https://github.com/Unleash/unleash-proxy-client-js
